src/ucp/wireup/wireup_ep.c

```diff
@@ -647,14 +647,17 @@ ucs_status_t ucp_wireup_ep_connect_to_sockaddr(uct_ep_h uct_ep,
     return status;
 }
 
-void ucp_wireup_ep_set_next_ep(uct_ep_h uct_ep, uct_ep_h next_ep)
+void ucp_wireup_ep_set_next_ep(uct_ep_h uct_ep, uct_ep_h next_ep,
+                               int is_owner, int is_local_connected)
 {
     ucp_wireup_ep_t *wireup_ep = ucp_wireup_ep(uct_ep);
 
     ucs_assert(wireup_ep != NULL);
     ucs_assert(wireup_ep->super.uct_ep == NULL);
-    wireup_ep->flags |= UCP_WIREUP_EP_FLAG_LOCAL_CONNECTED;
-    ucp_proxy_ep_set_uct_ep(&wireup_ep->super, next_ep, 1);
+    if (is_local_connected) {
+        wireup_ep->flags |= UCP_WIREUP_EP_FLAG_LOCAL_CONNECTED;
+    }
```

Comment:
why need this? can just call `ucp_proxy_ep_set_uct_ep` instead like it is done now

Reply:
I wanted to use `ucp_wireup_ep_set_next_ep()` akways instead of:
```C
if (local_conencted) {
    ucp_wireup_ep_set_next_ep();
} else {
    ucp_proxy_ep_set_uct_ep();
}
```